addons/netfox/serialization/value-to-bytes.gd

```diff
@@ -0,0 +1,202 @@
+extends Node
+class_name ValueToBytes ##Mayhaps needs to become Autoload so as to use the logger.
```

Comment:
Correct me if I'm wrong, but doesn't [`encode_var`](https://docs.godotengine.org/en/stable/classes/class_packedbytearray.html#class-packedbytearray-method-encode-var) implement the same?

Reply:
I haven't tried it thinking it was like `bytes_to_var`
Let me give it a try, because if true, that entire class is needless :+1:

Reply:
I am reading the godot code below because it seems `encode_var` includes the type at the header (so a `u_int32` integer is 8 bytes instead of 4 bytes as in this PR)
```cpp
if (buf) {
		encode_uint32(header, buf);
		buf += 4;
	}
	r_len += 4;

	switch (p_variant.get_type()) {
		case Variant::NIL: {
			//nothing to do
		} break;
		case Variant::BOOL: {
			if (buf) {
				encode_uint32(p_variant.operator bool(), buf);
			}

			r_len += 4;

		} break;
		case Variant::INT: {
			if (header & HEADER_DATA_FLAG_64) {
				//64 bits
				if (buf) {
					encode_uint64(p_variant.operator int64_t(), buf);
				}

				r_len += 8;
			} else {
				if (buf) {
					encode_uint32(p_variant.operator int32_t(), buf);
				}

				r_len += 4;
			}
		} break;

```

Reply:
ok so the header for strings (`NodePath` atm) is 12. Let me test this PR with Strings and PackedTypedArrays.
Because if this PR works with them (hadn't tried), then this PR is good. Otherwise, this custom serialization is needless, as it is automatically done by Godot's internals on sending the RPC.

Reply:
I tested it, and it seems I forgot to take into account the string size, and PackedTypeArray sizes, so they are not serialized properly. For both, I would have to add code which is about serializing arrays, so the header would include the byte length of the content, alongside 1 bit for what type, in `value-to-bytes.gd`

I know you are against this, so I have to ask: For GDScript serialization to be superior to Godot's Internals, it would have to be written in GDExtension. Assuming I were to make such a PR, would you accept it? The code in `RollbackSynchronizer` will stay the same as this PR's.

>If it's size, then we can probably get away with much simpler solutions, like not transmitting the input property names, and just putting the values into an array.

It sounds an easy PR given it is part of #264 , but the real bandwidth saver is diff states, not just omitting some strings. Only with diff states, I can see custom serialization be overkill.
So, if I make a PR which adds diff states without serialization included (so, half of #264 lol), how do you suggest I solve the problem of the first full state, as explained in #264?
1. Reliable RPC for the first full state, and queue/cache the diff states until the first state is received (skim the code of #267 )
2. Authority sends full states (unreliable) until the client sends an ack that he has received a full state, then authority switches to sending diff states
3. Project settings option where the user puts the amount of diff states to send in fullly (e.g. 40) before sending diff states, so kinda like input redundancy